### ISSUE TITLE
Service returns bad entities for correct search

### ISSUE THREAD

**Issue by althaus** [meilisearch/meilisearch-symfony#56]
Hi,

I've encountered another issue using the SearchService. The service returned different entities than the search client. I've debugged through the code and found this issue:

The search method has this line

~~~php
$ids = $this->engine->search($query, $this->searchableAs($className), $requestOptions);
~~~

which fetches the results from the client. The data is then used in the loop to gather the primary keys:

~~~php
            if (in_array($className, $this->aggregators, true)) {
                $entityClass = $className::getEntityClassFromObjectID($objectID);
                $id = $className::getEntityIdFromObjectID($objectID);
            } else {
                $id = $objectID;
                $entityClass = $className;
            }
~~~

In my case the code went into the `else` part and $objectID contained the **complete** data set. So it was an array and no plain literal. I've made the following change to get it working.

~~~php
$id = is_array($objectID) ? $objectID['objectID'] : $objectID;
~~~

Did I miss something? The code doesn't look as it could work at all?

Kind regards
Matthias

**Comment by curquiza** [comment]
Hello Matthias!
Thanks for this report!
I've checked the code, there are indeed some improvements we need to do!
The external contributor who did the code copy the Algolia repository, and here is the corresponding method -> https://github.com/algolia/search-bundle/blob/e1ba3243c6c520d4b6692381cf19c160b72fd298/src/Services/AlgoliaSearchService.php#L204-L239
However, he did not exactly the same methods as Algolia, especially because Algolia is not exactly like MeiliSearch
This leads to some errors like the one you found out and we missed it, sorry for this! I'm going to open a PR, it would be awesome if you can review it 😄

**Comment by althaus** [comment]
I'd be happy to help and test a fix.

I've just wondered if I missed some documentation as there is no real example in the wiki.

**Comment by curquiza** [comment]
The PR is ready for review #57 🚀

**Comment by baumanno** [comment]
I believe this issue can be closed, as #91 has been merged. Reading through the issue description, it sounds exactly like the bug adressed by that PR.

**Comment by curquiza** [comment]
Thanks @baumanno! I close this!
